juriscraper/lib/date_utils.py

```diff
@@ -154,3 +154,17 @@ def is_first_month_in_quarter(month):
     :return: Whether that month is the first month in a quarter
     """
     return month in [1, 4, 7, 10]
+
+def fix_future_year_typo(future_date):
+    """Fix year typo, convert 2106 to 2016"""
+    year = str(future_date.year)
+    if year[2] == '0':
+        fixed_year = int(year[0] + year[2] + year[1] + year[3])
+        if fixed_year < future_date_threshold():
+            fixed_date = '%d/%d/%s' % (future_date.month, future_date.day, fixed_year)
+            return datetime.datetime.strptime(fixed_date, "%m/%d/%Y").date()
+    return False
```

Comment:
Sure, by all means fix it up if there is a vulnerability! Sorry about that!  You can commit the changes yourself, or if you need me to push them up to that PR I can.

Sent from a phone

> On Aug 2, 2016, at 6:41 AM, Mike Lissner <notifications@github.com> wrote:
> 
> In juriscraper/lib/date_utils.py:
> 
> > @@ -154,3 +154,17 @@ def is_first_month_in_quarter(month):
> >      :return: Whether that month is the first month in a quarter
> >      """
> >      return month in [1, 4, 7, 10]
> > +
> > +def fix_future_year_typo(future_date):
> > +    """Fix year typo, convert 2106 to 2016"""
> > +    year = str(future_date.year)
> > +    if year[2] == '0':
> > +        fixed_year = int(year[0] + year[2] + year[1] + year[3])
> > +        if fixed_year < future_date_threshold():
> > +            fixed_date = '%d/%d/%s' % (future_date.month, future_date.day, fixed_year)
> > +            return datetime.datetime.strptime(fixed_date, "%m/%d/%Y").date()
> > +    return False
> We should probably just make it return a date no matter what -- either a fixed date or the original date. That way, functions that call fix_future_date() can just expect a good value out the other side.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>